test_conformance/api/test_clone_kernel.cpp

```diff
@@ -351,51 +466,467 @@ int test_clone_kernel(cl_device_id deviceID, cl_context context, cl_command_queu
+            != 0)
+        {
+            test_fail("test_exec_enqueue_helper failed for cloneKernel_2 on "
+                      "retry.\n");
+        }
+
+        clSVMFree(context, svmPtr_srcKernel);
+        clSVMFree(context, svmPtr_srcKernel_1);
+        clSVMFree(context, svmPtr_cloneKernel_1);
+        clSVMFree(context, svmPtr_cloneKernel_2);
+        clSVMFree(context, pBuf);
+
+        return TEST_PASS;
     }
+    else
+    {
+        return TEST_SKIPPED_ITSELF;
+    }
+}
+
+int test_empty_enqueue_helper(cl_command_queue queue, cl_kernel* srcKernel)
+{
+    cl_int error;
+    size_t ndrange1 = 1;
+
+    // enqueue - srcKernel
+    error = clEnqueueNDRangeKernel(queue, *srcKernel, 1, NULL, &ndrange1, NULL,
+                                   0, NULL, NULL);
+    test_error(error, "clEnqueueNDRangeKernel failed");
 
-    if (((float*)pbufRes)[3] != sa.f)
+    error = clFinish(queue);
+    test_error(error, "clFinish failed");
+
+    return TEST_PASS;
+}
+
+int test_cloned_kernel_empty_args(cl_device_id deviceID, cl_context context,
+                                  cl_command_queue queue, int num_elements)
+{
+    cl_int error;
+    clProgramWrapper program;
+    clKernelWrapper srcKernel;
+
+    // Create srcKernel to test with
+    error = create_single_kernel_helper(context, &program, &srcKernel, 1,
+                                        clone_kernel_test_kernel,
+                                        "test_kernel_empty");
+    test_error(error,
+               "Unable to create srcKernel for test_cloned_kernel_empty_args");
+
+    // enqueue - srcKernel
+    if (test_empty_enqueue_helper(queue, &srcKernel) != 0)
     {
-        test_error( error, "clCloneKernel test failed. Failed to clone float type argument." );
-        return -1;
+        test_fail("test_empty_enqueue_helper failed for srcKernel.\n");
     }
 
-    if (((int*)pbufRes)[6] != write_val)
+    // clone the srcKernel
+    clKernelWrapper cloneKernel_1 = clCloneKernel(srcKernel, &error);
+    test_error(error, "clCloneKernel failed for cloneKernel_1");
+
+    if (test_empty_enqueue_helper(queue, &cloneKernel_1) != 0)
     {
-        test_error( error, "clCloneKernel test failed.  Failed to clone cl_mem argument." );
-        return -1;
+        test_fail("test_empty_enqueue_helper failed for cloneKernel_1.\n");
     }
 
-    if (bimg)
+    // enqueue - srcKernel again
+    if (test_empty_enqueue_helper(queue, &srcKernel) != 0)
     {
-        error = test_image_arg_shallow_clone(deviceID, context, queue, num_elements, pbufRes, bufOut);
-        test_error( error, "image arg shallow clone test failed." );
+        test_fail("test_empty_enqueue_helper failed for srcKernel on retry.\n");
     }
 
-    if (bdouble)
+    return TEST_PASS;
+}
+
+int test_svm_enqueue_helper(cl_context context, cl_command_queue queue,
+                            cl_int* svmPtr_Kernel, cl_kernel* srcKernel,
+                            cl_int* value)
```

Comment:
See above - do we need another helper here, or can we reuse the previous helper from clSetKernelExecInfo?